tensorrt_llm/_torch/pyexecutor/py_executor.py

```
@@ -1257,6 +1262,29 @@ def _balance_adp_requests(self, context_requests: list[LlmRequest],
+                             generation_requests: list[LlmRequest]):
+        if not self.attention_tp_enable_waiting:
+            return context_requests
+
+        waited_context_requests = []
+        stop_waiting = False
+        num_scheduled_ctx_tokens = sum(
+            len(ctx_req.get_tokens(0)) for ctx_req in context_requests)
+        num_scheduled_gen_tokens = sum(
+            len(gen_req.get_tokens(0)) for gen_req in generation_requests)
+        num_scheduled_tokens = num_scheduled_ctx_tokens + num_scheduled_gen_tokens
+        stop_waiting = self.tp_ctx_waiting_iters_count >= self.attention_tp_timeout_iters or num_scheduled_tokens >= self.attention_tp_max_token_ratio * self.max_num_tokens
```

Ruff (0.12.2)

Line too long (172 > 120)

(E501)